changelog/6311.bugfix.rst

```diff
@@ -0,0 +1,2 @@
+Fixed bugs when working with a coordinate frame where the observer is specified in `~sunpy.coordinates.frames.HeliographicStonyhurst` and Cartesian representation, which is equivalent to Heliocentric Earth Equatorial (HEEQ).
+Now, the observer will always be converted to spherical representation.
```

Comment:
```suggestion
Now, the observer will always be converted to spherical representation when the coordinate is created.
```
If I'm interpreting this correctly, this is a slightly more helpful message.

Reply:
Edited